src/hooks/JuiceTheme.ts

```diff
@@ -19,32 +17,41 @@ const flattenNestedObject = (
     }
   }, {})
 
-export function useJuiceTheme(storageKey = 'jb_theme'): ThemeContextType {
-  const initialThemeOption =
-    (localStorage && (localStorage.getItem(storageKey) as ThemeOption)) ||
-    ThemeOption.light
+const userPrefersDarkMode = (): boolean => {
+  return window.matchMedia?.('(prefers-color-scheme: dark)')?.matches ?? false
+}
 
-  const [currentThemeOption, setCurrentThemeOption] =
-    useState<ThemeOption>(initialThemeOption)
+const getInitialThemeOption = (storageKey: string) => {
+  const storedThemeOption = localStorage.getItem(storageKey)
+  if (storedThemeOption) {
+    return storedThemeOption as ThemeOption
+  }
 
-  const [isDarkMode, setIsDarkMode] = useState<boolean>(
-    initialThemeOption === ThemeOption.dark,
+  return userPrefersDarkMode() ? ThemeOption.dark : ThemeOption.light
+}
+
+const setRootVarsForThemeOption = (themeOption: ThemeOption) => {
+  Object.entries(flattenNestedObject(juiceTheme(themeOption).colors)).forEach(
+    ([key, value]) =>
+      document.documentElement.style.setProperty('--' + key, value),
   )
 
-  const setRootVarsForThemeOption = (themeOption: ThemeOption) => {
-    Object.entries(flattenNestedObject(juiceTheme(themeOption).colors)).forEach(
-      ([key, value]) =>
-        document.documentElement.style.setProperty('--' + key, value),
+  Object.entries(juiceTheme(themeOption).radii).forEach(([key, value]) => {
+    if (!value) return
+    document.documentElement.style.setProperty(
+      '--radius-' + key,
+      value.toString(),
     )
+  })
+}
 
-    Object.entries(juiceTheme(themeOption).radii).forEach(([key, value]) => {
-      if (!value) return
-      document.documentElement.style.setProperty(
-        '--radius-' + key,
-        value.toString(),
-      )
-    })
-  }
+export function useJuiceTheme(storageKey = 'jb_theme'): ThemeContextType {
+  const initialThemeOption = getInitialThemeOption(storageKey)
```

Comment:
You need to make sure this is wrapped in a browser only ctx, or server will crash on attempting to read `window`

Reply:
@wraeth-eth great catch!!!! Checking for presence of `window` object now:

https://github.com/jbx-protocol/juice-interface/pull/2289/files#diff-0072acd67b85ccdf7fa522877db68c7dbeb8e13af574b56360d96626eda5f05aR21-R23

Do you think this is sufficient?